src/compact/compact.js

```diff
@@ -0,0 +1,40 @@
+import { forEach } from "../for-each/for-each"
+import { is } from "../is/is"
+import { map } from "../map/map"
+import { filter } from "../filter/filter"
+import { curry } from "../curry/curry"
+
+const _compactObject = source => {
+  const result = { ...source }
+
+  forEach(key => {
+    if (!is(result[key])) {
+      delete result[key]
+    }
+  }, Object.keys(result))
+
+  return result
+}
+
+const _compactArray = source => filter(is, source)
+
+const _compact = source => {
+  const type = Array.isArray(source) ? "array" : typeof source
```

Comment:
There is a problem here with `typeof source`, when `null` is passed, the type is "object".

We already have a function, [`type`](https://github.com/asd-xiv/m/blob/master/src/type/type.js), that returns a more detailed and reliable description.

```js
import { type } from "../type/type"

const _compact = source => {
  const sourceType = type(source)

  switch (sourceType) {
    case "Array":
      return _compactArray(source)

    case "Object":
      return _compactObject(source)

    default:
      return source
  }
}
```